ui/src/components/Nodes/NodeHierarchyPanel.vue

```diff
@@ -0,0 +1,200 @@
+      <div v-if="selectedFlowCount">
+        <span>{{ `Flows (${selectedFlowCount})` }}</span>
+        <FeatherButton icon="Clear" @click="onClearFlows">
+          <FeatherIcon :icon="clearIcon"> </FeatherIcon>
+        </FeatherButton>
+      </div>
+      <div v-else>Flows</div>
+    </template>
+    <FeatherList class="category-list">
+      <FeatherListItem
+        v-for="flow of flowTypes"
+        :selected="isFlowSelected(flow)"
+        :key="flow"
+        @click="onFlowClick(flow)">
+        {{ flow }}
+      </FeatherListItem>
+    </FeatherList>
+  </FeatherExpansionPanel>
+  <FeatherExpansionPanel>
+    <template v-slot:title>
+      <div v-if="selectedLocationCount">
+        <span>{{ `Locations (${selectedLocationCount})` }}</span>
+        <FeatherButton icon="Clear" @click="onClearLocations">
+          <FeatherIcon :icon="clearIcon"> </FeatherIcon>
+        </FeatherButton>
+      </div>
+      <div v-else>Locations</div>
+    </template>
+    <FeatherList class="category-list">
+      <FeatherListItem
+        v-for="loc of locations"
+        :selected="isLocationSelected(loc)"
+        :key="loc.name"
+        @click="onLocationClick(loc)">
+        {{ loc.name }}
+      </FeatherListItem>
+    </FeatherList>
+  </FeatherExpansionPanel>
+</template>
+
+<script setup lang="ts">
+import ClearIcon from '@featherds/icon/action/Cancel'
+import { FeatherButton } from '@featherds/button'
+import { FeatherExpansionPanel } from '@featherds/expansion'
+import { FeatherIcon } from '@featherds/icon'
+import { FeatherList, FeatherListItem } from '@featherds/list'
+import { useStore } from 'vuex'
+import { Category, MonitoringLocation } from '@/types'
+
+const store = useStore()
+const clearIcon = ref(ClearIcon)
+const categories = computed<Category[]>(() => store.state.hierarchyModule.categories)
+const selectedCategories = computed<Category[]>(() => store.state.hierarchyModule.selectedCategories)
+const flowTypes = computed<string[]>(() => ['Ingress', 'Egress'])
+const selectedFlows = computed<string[]>(() => store.state.hierarchyModule.selectedFlows)
+
+const locations = computed<MonitoringLocation[]>(() => store.state.hierarchyModule.monitoringLocations)
+const selectedLocations = computed<MonitoringLocation[]>(() => store.state.hierarchyModule.selectedMonitoringLocations)
+const selectedCategoryCount = computed<number>(() => selectedCategories.value?.length || 0)
+const selectedFlowCount = computed<number>(() => selectedFlows.value?.length || 0)
+const selectedLocationCount = computed<number>(() => selectedLocations.value?.length || 0)
+
+const isCategorySelected = (cat: Category) => {
+  return selectedCategories.value.some(c => c.id === cat.id)
+}
+
+const isFlowSelected = (flow: string) => {
+  return selectedFlows.value.some(f => f === flow)
+}
+
+const isLocationSelected = (loc: MonitoringLocation) => {
+  return selectedLocations.value.some(x => x.name === loc.name)
+}
+
+const onClearCategories = () => {
```

Comment:
If there's an opportunity to start migrating to pinia in the future within this product, you can cut down on a lot of these boilerplate methods, and also move away from string-based events. Nothing to be done now though, just a comment for the futureeee. :)